src/hooks/use-site-details.tsx

```diff
@@ -87,20 +87,29 @@ function useDeleteSite() {
 
 	const deleteSite = useCallback(
 		async ( siteId: string, removeLocal: boolean ): Promise< SiteDetails[] | undefined > => {
-			const siteSnapshots = snapshots.filter( ( snapshot ) => snapshot.localSiteId === siteId );
-
```

Comment:
This is not really directly related to the issue but I noticed that we are checking for the `siteId` presence after we use it to locate `siteSnapshots` so I thought it made sense to move this below the check for the `siteId`.

Reply:
Good catch 👍